web/ui/src/Services/AppsController.js

```diff
@@ -292,34 +267,108 @@
                 $scope.templates.data = utils.mapToArr(templates);
             });
         }
-       
-        function fetchInternalServiceInstances(app) {
-            return resourcesFactory.v2.getInternalServiceInstances(app.id).then( data => {
-                app.instances = data;
+
+        function refreshApps() {
+            return fetchApps()
+                .then(fillOutApps)
+                .then(fetchServicesHealth)
+                .then(fetchInternalServicesHealth)
+                .then(() => touch());
+        }
+
+        function fetchApps() {
+            return $q.all([fetchInternalService(), fetchServices()])
+                .then(results => {
+                    $scope.apps = [];
+
+                    var isvc = results[0];
+                    $scope.apps.push(isvc);
+
+                    var services = results[1];
+                    services.forEach(s => {
+                        $scope.apps.push(s);
+                    });
+                });
+        }
+
+        function fetchChildEndpoints(app) {
+            return resourcesFactory.v2.getServiceChildPublicEndpoints(app.id).then(data => {
+                app.publicEndpoints = data;
```

Comment:
In `Service.js`, public endpoints get their parent's `DesiredState` attached so that the publicEndpointLinkDirective knows if the link should be active or not. That is probably needed here.

`Service.js` also adds `Value` but that appears to only be used when listing exported service endpoints for adding public endpoints, so it can probably be skipped.